### DISCUSSION TITLE
no phonetic result with elasticsearch

### DISCUSSION THREAD

**Discussion by ghost** [schranz-search/schranz-search#189] [Q&A]
Maybe this question is irrelevant as I am no search expert.
I tested this library by following the standalone code from the getting started page with elasticsearch and everything worked perfectly even icu plugin with Arabic content but failed to get it work with phonetic plugin.
content "john doe"
search query "jon" no result returned

I can provide a repo of the test if needed

Thanks

**Comment by alexander-schranz** [comment]
Really thank you for testing, a reproducer would be nice. I'm not familiar with the phonetic plugin, does it work with the `query_string` `query` of elasticsearch? We are using that for query elasticsearch: https://github.com/schranz-search/schranz-search/blob/be611c1f89cef1ec5d105e8d2f7ec6fe7729a116/packages/seal-elasticsearch-adapter/src/ElasticsearchSearcher.php#LL84C72-L84C87

Or is it required with the phonetic plugin if that we still need to mark specific searchable fields that they use that plugin?

**Reply by ghost** [reply]
Here is [the repository](https://github.com/hichamtaoufikallah/schranz-search-demo) with the simple code + Dockerfile (icu+phonetic plugins) + docker-compose
Yeah the phonetic does work with `query_string query` but it does require some more settings please check this [article](https://spinscale.de/posts/2021-06-30-implementing-phonetic-search-with-elasticsearch.html), this [doc](https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis-phonetic-token-filter.html) and this [SO question](https://stackoverflow.com/questions/62883549/elastic-search-with-phonetic-search) from all those pages it seems that an encoder (metaphone, double_metaphone, soundex, refined_soundex, caverphone1, caverphone2, cologne, nysiis, koelnerphonetik, haasephonetik, beider_morse, daitch_mokotoff) should be configured

Again I am no expert of setting up elasticsearch :)
